### PR TITLE
Add Dockerfile and README for network images

### PR DESCRIPTION
**Pull request by GabyCT**
Add Dockerfile to create the network image currently used in all our networking testing. This Dockerfile will install all the dependencies that are required.
Add a README where it has the basic steps of building the image and verifying the image.

### CODE REVIEW

**Comment by chavafg** [comment]
qa-passed

**Comment by jodh-intel** [comment]
lgtm

File: tests/metrics/network/dockerfile_image/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Building the network image
```
Comment (rcaballeromx) [inline tests/metrics/network/dockerfile_image/README.md]: Please, do not use headings for numbered steps. Also use the imperative:
Build the network image with the following command:

File: tests/metrics/network/dockerfile_image/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Building the network image 
+
+```
+$ docker build -t cc-network .
+```
+
+# Verifying the network image
```
Comment (rcaballeromx) [inline tests/metrics/network/dockerfile_image/README.md]: Please use numbered steps. Also use the imperative:
Verify the network image with the following command:

File: tests/metrics/network/dockerfile_image/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+#
+# - a basic development environment
+# - runtime build dependencies and additional general tooling
+#
+RUN apt-get -y update && \
+    apt-get install -y build-essential && \
+    apt-get install -y curl
+
+# Install smem (Memory reporting tool)
+RUN apt-get install -y smem
```
Comment (egernst) [inline tests/metrics/network/dockerfile_image/Dockerfile]: I haven't looked out other Dockerfiles we maintain, and this is definitely a nitpick, but from reading Dockerfile best practices, they suggest doing all the apt-get's in one line, so for this it'd be:

RUN apt-get update && apt-get install -y \
  build-essential \
  curl \
  iperf3 \
  iperf \
  smem

At the very least, I'm not sure you need the -y for the initial apt-get update.

Comment (egernst) [inline tests/metrics/network/dockerfile_image/Dockerfile]: I'd be curious if others have input on this, though - it's more of a style thing.  @jodh-intel ?

Best practices link: https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/

Comment (jodh-intel) [inline tests/metrics/network/dockerfile_image/Dockerfile]: @egernst - agreed - let's minimise the number of layers.

File: tests/metrics/network/dockerfile_image/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+
+# XXX: named version
```
Comment (grahamwhaley) [inline tests/metrics/network/dockerfile_image/Dockerfile]: I don't know dockerfiles - but is that 'XXX' a hangover from something - needs updating or deleting?

**Comment by chavafg** [comment]
qa-passed

**Comment by chavafg** [comment]
qa-passed

File: tests/metrics/network/dockerfile_image/README.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# Networking image for Intel® Clear Containers performance metrics
+
+If you want to build the network image with the general tooling
+and system dependencies that we are using in our networking tests,
+just do the following:
```
Comment (rcaballeromx) [inline tests/metrics/network/dockerfile_image/README.md]: Please change the paragraph to:
In order to build the network image with the general tooling
 and system dependencies which are used in our networking tests, follow these steps:

**Comment by GabyCT** [comment]
@rcaballeromx changes applied thanks

**Comment by chavafg** [comment]
qa-passed

**Comment by chavafg** [comment]
qa-passed

**Comment by jodh-intel** [comment]
lgtm

**Pull request merged**